test/dotnet-msbuild.Tests/GivenDotnetPublishInvocation.cs

```diff
@@ -72,6 +72,23 @@ public void MsbuildInvocationIsCorrectForSeparateRestore(string[] args, string e
                    .Be($"{ExpectedPrefix} -nologo -target:Publish{expectedAdditionalArgs}");
         }
 
```

Comment:
I would suggest these additional test cases (mostly to guard against regressions in changes that I'm planning to make in the future):

- `dotnet build` then `dotnet publish -r <RID>` should fail due to missing target in assets file
- `dotnet build -r <RID>` then `dotnet publish -r <RID>` should succeed

Reply:
oops, yes :-)

Reply:
Done.